transactor/src/main/scala/io/mediachain/datastore/PersistentDatastore.scala

```diff
@@ -0,0 +1,92 @@
+package io.mediachain.datastore
+
+import java.util.Random
+import java.util.concurrent.LinkedBlockingDeque
+import com.amazonaws.AmazonServiceException
+import org.slf4j.{Logger, LoggerFactory}
+import io.mediachain.multihash.MultiHash
+
+class PersistentDatastore(config: PersistentDatastore.Config)
+  extends BinaryDatastore with AutoCloseable with Runnable {
+  
+  val logger = LoggerFactory.getLogger(classOf[PersistentDatastore])
+  val dynamo = new DynamoDatastore(config.dynamo)
+  val rocks = new RocksDatastore(config.rocks)
+  val random = new Random
+  val maxBackoffRetry = 60 // second
+  val queue = new LinkedBlockingDeque[MultiHash]
+  val writer = new Thread(this, "PersistentDataStore#write")
```

Comment:
i definitely see what yr getting at with this, though i feel pretty strongly that this should be called from the outside. this is a very surprising side effect to someone from the outside world. if anything, i could see there being an instance method `start()` or something like this to make it more clear so you'd have

```scala
val pds = new PersistentDatastore(config)
pds.start
```